### PR TITLE
Update exception message

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
This is an automated comment for commit a1eb27ce3038df0921d6667e65074168776988dc with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51440/a1eb27ce3038df0921d6667e65074168776988dc/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Disks/IO/CachedOnDiskReadBufferFromFile.cpp [hunk 1]
```diff
@@ -940,7 +941,8 @@ bool CachedOnDiskReadBufferFromFile::nextImplStep()
         // We don't support implementation_buffer implementations that use nextimpl_working_buffer_offset.
         chassert(implementation_buffer->position() == implementation_buffer->buffer().begin());
 
-        size = implementation_buffer->buffer().size();
+        if (result)
+            size = implementation_buffer->buffer().size();
```
Comment (SmitaRKulkarni) [inline src/Disks/IO/CachedOnDiskReadBufferFromFile.cpp:945]: If `!result` , then previous value will be used in the log & incrementing ProfileEvents too. Is this fine ?
https://github.com/ClickHouse/ClickHouse/blob/a1eb27ce3038df0921d6667e65074168776988dc/src/Disks/IO/CachedOnDiskReadBufferFromFile.cpp#L900

**Review by SmitaRKulkarni — COMMENTED** [review]
Rest all LGTM

**Comment by kssenii** [comment]
> Stress test (debug) — Killed by signal

https://github.com/ClickHouse/ClickHouse/issues/51289

> Integration tests (release) [2/4] — fail: 1

test_storage_kafka/test.py::test_kafka_formats_with_broken_message

Stateless tests not related as well

**Pull request merged**